### ISSUE TITLE
Does `preact.render` support multiple components within a single parent node?

### ISSUE THREAD

**Issue by natevw** [preactjs/preact#1665]
While debugging a Preact 8.4.2 issue (which turns out to be recycling related…) I accidentally started reading the Preact X `render` code instead:

```
export function render(vnode, parentDom, replaceNode) {
	if (options.root) options.root(vnode, parentDom);
	let oldVNode = parentDom._prevVNode;
	vnode = createElement(Fragment, null, [vnode]);

	let mounts = [];
	diffChildren(
		parentDom,
		replaceNode ? vnode : (parentDom._prevVNode = vnode),
		oldVNode,
// …snip…
```

What struck me was the `parentDom._prevVNode` logic, which implies that within a given DOM element only **one** of its children should ever be a preact component?

I have a situation where I pre-create elements (at exact offsets amongst text nodes within a parent `<p>` element) and then use Preact to actually fill in the contents of said elements, so this gives me concern.

What I'm doing is very very roughly equivalent to:

```
function myUpdate(contents) {
  let container = document.getElementById('#my_container'),
      el1 = container.querySelector('span.componentA'),
      el2 = container.querySelector('span.componentB');
  preact.render(html`<${MyComponent} content=${contents['a']} />`, container, el1);
  preact.render(html`<${MyComponent} content=${contents['b']} />`, container, el2);
}
```

…but it looks like preact might get confused since my `container` element would then have conflicting values stored on it for `._prevVNode` due to the two different renders? Can I repeatedly call `myUpdate` and expect both instances of `MyComponent` to update correctly?

**Comment by natevw** [comment]
To be fair, I should point out that React just simply doesn't support this at all. The current version of https://reactjs.org/docs/react-dom.html#render reads:

> ReactDOM.render() controls the contents of the container node you pass in.

React makes you always have `.obligatory-wrapper > .rendered-vnode`. Preact's API tempts one to think they can avoid needing the extra wrapper that React always needs, but between the assumptions seemingly made by the Preact X code above, and some recycler-related "fun surprises" in Preact 8, I'm realizing that Preact may simply be the same limitation, except that I have to explicitly pass the `.rendered-vnode` instead of it finding it for me.

(FWIW the Preact 8.4.2 bug was that even if I gave `preact.render()` a specific element to render into, it would toss that element and replace it with a recycled one instead which caused some mayhem on the D3 side of the logic. I've updated the code to add an extra wrapper element both to set a clearer boundary between D3 and Preact, and so if I'm right about the above I now always have an `.obligatory-wrapper` dedicated just to one node instead of the parent being potentially shared.)
